### PR TITLE
Adding ciphers resource documentation

### PR DESCRIPTION
**Pull request by the2hill**
**Quality audit reminder**

Doc team: While logged in using your O365 account, complete the quality checklist at http://bit.ly/2hwvBKX. Use the checklist for significant doc changes or additions submitted by you or a non-doc staff contributor.

### CODE REVIEW

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-2c8b75e26d/docs/cloud-load-balancers/v2/)
* [`internal-api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-2c8b75e26d/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-2c8b75e26d/docs/cloud-load-balancers/v1/)

File: api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+==============
+Ciphers 
+==============
+
+.. contents::
+   :local:
+   :depth: 1
+
+Users can view the ciphers defined on the load balancer. Ciphers
+are set utilizing the CiphersProfile field when creating SSL Termination
```
Comment (cyrichardson) [inline api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst]: ``cipherProfile``

File: api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+for the Load Balancer. Utilizing the ciphers resource we can quickly
```
Comment (cyrichardson) [inline api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst]: for the load balancer. (lower case)
Utilizing the ciphers resource, you can quickly

File: api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+and easily determine what ciphers the Load Balancer adheres to.
```
Comment (cyrichardson) [inline api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst]: load balancer (lower case)

File: api-docs/cloud-load-balancers-v1/api-reference/methods/get-list-ciphers-v1.0-account-loadbalancers.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Response
+--------
+
+
+The following table shows the body parameters for the response:
+
++--------------------------+-------------------------+-------------------------------+
+|Name                      |Type                     |Description                    |
++==========================+=========================+===============================+
+|cipherList                   |String                   |A ``loadBalancers``            |
```
Comment (cyrichardson) [inline api-docs/cloud-load-balancers-v1/api-reference/methods/get-list-ciphers-v1.0-account-loadbalancers.rst]: Line up the "|" character with those above it for the table.

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`internal-api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-82d377f582/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-82d377f582/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-82d377f582/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
Thank you for the review @cyrichardson. How can I see what failed with the Strider tests?

**Comment by cyrichardson** [comment]
@the2hill Can you access this? https://build.developer.rackspace.com/

**Comment by cyrichardson** [comment]
It is showing a successful strider build. Let me see if our tools person is in and has a clue - @nimbinatus ??

**Comment by cyrichardson** [comment]
There are some warnings, but no errors. Maybe a network glitch or somethng?

**Comment by the2hill** [comment]
@cyrichardson Not authorized :( I have to look again, it also seems to not be showing the resource in the legend. I may have to push another update and maybe that'll fix the UI here.

**Comment by nimbinatus** [comment]
@the2hill, I'm adding you to the build view team. Give me a couple minutes.

@cyrichardson, I'll retrigger the build. It's a very sporadic (once in a blue moon) error that we haven't tracked down.

**Comment by cyrichardson** [comment]
@the2hill Good idea... just push another small change.

**Comment by cyrichardson** [comment]
Thanks, @nimbinatus!

**Comment by the2hill** [comment]
Thank you @nimbinatus!

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`internal-api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-82d377f582/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-82d377f582/docs/cloud-load-balancers/v2/)

**Comment by nimbinatus** [comment]
@the2hill and @cyrichardson all fixed!

**Comment by the2hill** [comment]
@nimbinatus Thank you! @cyrichardson I'm a bit new to this doc format. I'm curious why my resource isn't showing up. I have the .rst in the 'api-reference' dir and I'm not seeing anywhere else it might need to be defined. Any tips are appreciated!

**Comment by cyrichardson** [comment]
@the2hill Just need to add the new resource to the index file to get it to show up in left navigation pane.

**Comment by the2hill** [comment]
Oh, I take that back... my apologies. I have a few things to add :)

**Comment by the2hill** [comment]
I see that now :) Thank you!

**Comment by cyrichardson** [comment]
@the2hill https://github.com/rackerlabs/docs-cloud-load-balancers/blob/master/api-docs/cloud-load-balancers-v1/api-reference/index.rst

**Comment by the2hill** [comment]
We still need to decide where the 'footnote' for the actual profile definitions are going to live. So this may hangout for a bit until then. Thanks again for the help/reviews!

**Comment by cyrichardson** [comment]
OK, @the2hill! Glad to be able to help! Thank you for your doc contributions!

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`internal-api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-329f0c7bc0/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-329f0c7bc0/docs/cloud-load-balancers/v1/)
* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-329f0c7bc0/docs/cloud-load-balancers/v2/)

File: api-docs/cloud-load-balancers-v1/api-reference/methods/get-list-ciphers-v1.0-account-loadbalancers.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+Response
+--------
+
+
+The following table shows the body parameters for the response:
+
++--------------------------+-------------------------+-------------------------------+
+|Name                      |Type                     |Description                    |
++==========================+=========================+===============================+
+|cipherList                |String                   |A ``loadBalancers``            |
```
Comment (MadhaviLReddy) [inline api-docs/cloud-load-balancers-v1/api-reference/methods/get-list-ciphers-v1.0-account-loadbalancers.rst]: The Description value says "A loadBalancers object.", can we add little more explanation here? For ex: List of comma separated cipher names that are defined on the loadbalancer.

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-bfd996427f/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-bfd996427f/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
@cyrichardson How would I go about adding link backs to a 'reference' from within a 'method' rst file?

**Comment by the2hill** [comment]
example, from within the method/ssltermination...rst make a link back inside of a **note to  api-reference/ciphers.rst file.

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-db7ce615e1/docs/cloud-load-balancers/v1/)
* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-db7ce615e1/docs/cloud-load-balancers/v2/)

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-9f3c066de9/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-9f3c066de9/docs/cloud-load-balancers/v1/)

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-bc905c9b75/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-bc905c9b75/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
Hrmm, it doesnt like my table. I'll give it another go in the morning.

File: api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+CiphersProfiles are cipher suites defined by the service and versioned by date.
+As security concerns change cipher profiles will be added with an updated
+cipher list. Below is a matrix of the available profiles and their associated
+cipher lists.
+
++--------------------------------------+---------+------------------------+
+|  CipherProfiles                      | default | CLBCipherPolicy2017-08 |
++======================================+=========+========================+
++--------------------------------------+---------+------------------------+
```
Comment (cyrichardson) [inline api-docs/cloud-load-balancers-v1/api-reference/ciphers.rst]: I think it doesn't like these 2 heading rows.

**Comment by cyrichardson** [comment]
You can see these messages here: https://build.developer.rackspace.com/rackerlabs/docs-cloud-load-balancers/job/597ab00788731c0100eb02db

Let me know if you don't have access to see that.

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-fbb1906645/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-fbb1906645/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
I got those same errors, I think I can resolve that one too if this new one doesn't work. Thanks for the link, that does indeed help!

**Comment by the2hill** [comment]
Ah, not authorized for the link though :)

**Comment by the2hill** [comment]
okay, cool, i think the way i have it now might work. I don't know why its not splitting the headers though :/ Ill try a few more things. At least it's parsing now.

**Comment by the2hill** [comment]
Hmm, not sure what else I can do to change it.. In either case, I think I'm content with the way it is for now. When we add more profiles I would like to expand on it and make it easier to edit. So, I think this is read for review and we should be solid for merge soon after.

**Comment by cyrichardson** [comment]
I think you can only have the Ciphers heading row or the CipherProfiles heading row, but not both. I'll ask to get you access to the Strider dashboard so you can see errors there.

**Comment by the2hill** [comment]
I was trying to do something similar to the example: http://openalea.gforge.inria.fr/doc/openalea/doc/_build/html/source/sphinx/rest_syntax.html

which I suppose is giving me what the docs show. It's just not really the table look I was going for, but the design it should have. So, this works for now unless others object :)

**Comment by cyrichardson** [comment]
It looks fine at the staging link to me. Great. @the2hill!

**Comment by cyrichardson** [comment]
@the2hill Just checking in to see if you have a date for all the cipher addition PRs (for the docs JIRA board). Thanks!

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-2c78b7a16c/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-2c78b7a16c/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
Sorry, I did not see this. A few things came up and we've had to make some changes. We're planning for Friday the 8th, I believe.

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-8edb9a0892/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-8edb9a0892/docs/cloud-load-balancers/v1/)

**Comment by cyrichardson** [comment]
No worries, @the2hill. I will be out of the office starting on 9/1 and not back until 9/14. Please ask for any reviews you need in the #docs slack channel.

**Comment by the2hill** [comment]
Thank you!

**Comment by filler** [comment]
Hey @cyrichardson.  What does it take in order to get this change out?  When you hit merge, does a CI process kick off and the release to production docs in front of customers?

**Comment by cyrichardson** [comment]
@filler Yes! That's exactly what should happen.

**Comment by filler** [comment]
Also leaving a comment here that CLB needs to extend Release Notes to reflect the upcoming ciphers feature. cc @the2hill

**Comment by rackernexus** [comment]
HULK PREVIEW

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-8edb9a0892/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-8edb9a0892/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
@filler done, not sure if we want to add more details, but its up for review.

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-9d2af8377a/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-9d2af8377a/docs/cloud-load-balancers/v1/)

**Comment by the2hill** [comment]
Forgot index..

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-711e524ee2/docs/cloud-load-balancers/v2/)
* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-711e524ee2/docs/cloud-load-balancers/v1/)

**Comment by rackernexus** [comment]
Your content preview is now ready:

* [`api-docs/cloud-load-balancers-v1`](https://staging.developer.rackspace.com/build-a105b29fd8/docs/cloud-load-balancers/v1/)
* [`api-docs/cloud-load-balancers-v2`](https://staging.developer.rackspace.com/build-a105b29fd8/docs/cloud-load-balancers/v2/)

**Comment by cody-somerville** [comment]
PR #188 includes the contents of this PR (with attribution retained) and thus supersedes it. Thank you for your work on this @the2hill - much appreciated!

**Pull request closed without merging**